File: clase1/src/main/java/com/clase1/service/ClienteService.java
package com.clase1.service;

import java.util.List;

import com.clase1.entity.Cliente;
import com.clase1.exception.ClienteNotFoundException;

public interface ClienteService {

	public List<Cliente> getAll();
	public Cliente getById(Long id) throws ClienteNotFoundException;
	public Cliente save(Cliente cliente);
	public Cliente delete(Long id) throws ClienteNotFoundException;
	
}

File: front/src/app/services/cliente.service.ts
import { Injectable } from '@angular/core';
import { Cliente } from '../models/cliente';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable()
export class ClienteService {

    private _uri : string = 'http://localhost:8099/clientes';

    constructor(
        private _httpClient : HttpClient
    ) {}

    getAll() : Observable<Array<Cliente>> {
        return this._httpClient.get<Array<Cliente>>(this._uri);
    }

    getById(id : number) {
        return this._httpClient.get<Cliente>(this._uri + '/' + id);
    }

    save(cliente : Cliente) {
        return this._httpClient.post<Cliente>(this._uri, cliente);
    }

    delete(id : number) {
        return this._httpClient.delete<Cliente>(this._uri + '/' + id);
    }
}